hack/deploy_loki_using_observatorium_yaml.sh

```diff
@@ -0,0 +1,125 @@
+    exit
+  fi
+
+  if [ -z "$AWS_S3_SECRET_KEY" ]; then
+    echo "==> ERROR: Provide AWS secret_key using AWS_S3_SECRET_KEY env.variable or using $AWS_CREDENTIALS_FILE_NAME file"
+    exit
+  fi
+
+  echo "
+Using AWS S3 configuration for loki bucket::
+-==-=-=-=-=-=-
+AWS S3 end-point: $AWS_S3_END_POINT
+AWS S3 region: $AWS_S3_REGION
+
+bucket name: $AWS_S3_LOKI_BUCKET_NAME
+access_key: $AWS_S3_ACCESS_KEY
+secret_key: $AWS_S3_SECRET_KEY
+
+(Note: Use environment variables as needed to change defaults)
+"
+
+
+}
+
+check_oc_works() {
+  OC_WORKS=$(oc get projects > /dev/null 2>&1 ; echo $?)
+  if ! [ "$OC_WORKS" == "0" ]; then
+    echo "==> ERROR: oc command not working,  make sure you are logged into OCP cluster and that oc command works"
+    exit
+  else
+    echo "OK, OCP cluster working"
+  fi
+}
+
+delete_project_if_exists() {
+  PROJECT=$(oc get project | grep "$1")
+  if [ -n "$PROJECT" ]; then
+    echo "Deleting $1 project/namespace"
+    oc delete project "$1"
+    while : ; do
+      PROJECT=$(oc get project | grep "$1")
+      if [ -z "$PROJECT" ]; then break; fi
+      sleep 1
+    done
+  fi
+}
+
+create_project() {
+  oc new-project "$1" > /dev/null 2>&1
+  echo "Done"
+}
+
+get_observatorium_template() {
+  rm -f observatorium-logs-template.yaml
+  wget -nv https://raw.githubusercontent.com/rhobs/configuration/main/resources/services/observatorium-logs-template.yaml
+}
+
+create_s3_secret() {
+  rm -f $LOKI_S3_SECRET_NAME.yaml
+  cat > $LOKI_S3_SECRET_NAME.yaml << EOF
+apiVersion: v1
+kind: Secret
+data:
+  endpoint: $(echo "$AWS_S3_END_POINT" | base64)
+  aws_region: $(echo "$AWS_S3_REGION" | base64)
+  bucketnames: $(echo "$AWS_S3_LOKI_BUCKET_NAME" | base64)
+  bucket: $(echo "$AWS_S3_LOKI_BUCKET_NAME" | base64)
+  aws_access_key_id: $(echo "$AWS_S3_ACCESS_KEY" | base64)
+  aws_secret_access_key: $(echo "$AWS_S3_SECRET_KEY" | base64)
+metadata:
+  name: $LOKI_S3_SECRET_NAME
+  namespace: $LOKI_PROJECT_NAME
+EOF
+    echo "Done"
+}
+
+deploy_loki() {
+  echo "Deploying Loki s3 secret from = $LOKI_S3_SECRET_NAME.yaml"
+  oc apply -f $LOKI_S3_SECRET_NAME.yaml
+  echo "Deploying Loki manifests with ingester replicas = $LOKI_INGESTER_REPLICAS into project $LOKI_PROJECT_NAME"
+  oc process -f observatorium-logs-template.yaml  -p LOKI_INGESTER_REPLICAS="$LOKI_INGESTER_REPLICAS" -p NAMESPACE=$LOKI_PROJECT_NAME -p LOKI_S3_SECRET=$LOKI_S3_SECRET_NAME | oc apply -f -
+  echo "Done"
+}
+
+main () {
+  echo "==> Check that oc works"
+  check_oc_works
+  echo "==> AWS credentials for Loki Bucket"
+  get_aws_credentials
+  echo "==> Create object storage secret file"
+  create_s3_secret
+  echo "==> Get observatorium template"
+  get_observatorium_template
+  echo "==> Delete project $LOKI_PROJECT_NAME"
+  delete_project_if_exists $LOKI_PROJECT_NAME
+  echo "==> Create project $LOKI_PROJECT_NAME"
+  create_project $LOKI_PROJECT_NAME
+  echo "==> Deploy loki"
+  deploy_loki
+}
+
+main
+oc get pods
```

Comment:
Suggest to drop the last command, it's the users choice to look after the pods. If the intention is to create a notion of waiting until everything deploys, suggestion here is to use:
```
oc rollout status deployment ...
oc rollout status statefulset ...
```
See for example: https://github.com/observatorium/loki-benchmarks/blob/master/run.sh#L58

Reply:
I dropped the last line ... agree this is not needed.